From 26509cf8f4585fcd751f3a498242d7cc85576026 Mon Sep 17 00:00:00 2001
From: Albin B <albin.bala.krishnan@intel.com>
Date: Tue, 8 Dec 2015 11:07:46 +0530
Subject: [PATCH 4/5] [EM/USB_PD]: Enable charging upon PS_RDY command receive

Setup charging parameters if power supply ready command received,
regardless of charging the state when it is in ready state.

Change-Id: Iaa9861cf61ed28dd6a96fd7ef90ef2805538d8ed
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-12466
Signed-off-by: Albin B <albin.bala.krishnan@intel.com>
Reviewed-on: https://android.intel.com:443/448733
---
 drivers/usb/typec/pd/sink_port_pe.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/typec/pd/sink_port_pe.c b/drivers/usb/typec/pd/sink_port_pe.c
index 31daab7..8042005 100644
--- a/drivers/usb/typec/pd/sink_port_pe.c
+++ b/drivers/usb/typec/pd/sink_port_pe.c
@@ -819,8 +819,9 @@ static void sink_handle_ready(struct sink_port_pe *sink)
 		schedule_work(&sink->request_timer);
 		goto ready_end;
 	}
-	if (sink->cur_state == PE_SNK_TRANSITION_SINK)
-		snkpe_setup_charging(sink);
+
+	/* enable charging if ps_rdy received */
+	snkpe_setup_charging(sink);
 
 ready_end:
 	sink->p.status = POLICY_STATUS_SUCCESS;
-- 
1.9.1

